packages/headless/src/database/mongodb/customer/index.ts

```diff
@@ -48,7 +48,10 @@ export class CustomerDatabase implements ICustomerDatabase {
   }
 
   async updateCustomer(customerId: string, customer: Customer): Promise<Customer | null> {
-    return await CustomerModel.findOneAndUpdate({ id: customerId }, { $set: customer }, { new: true }).lean().select('-password -_id').exec();
+    const updatedCustomer = await CustomerModel.findOneAndUpdate({ id: customerId }, { $set: customer }, { new: true }).lean().select('-password -_id').exec();
```

Comment:
please flatten an object before updating in database. otherwise some object might be overwritten.
for example your database has some customer like: 
`{
    phone: {
                primary: "1234", 
                secondary: "5768"
           }
}`

and you want to update only primary number like this:
`{
    phone: {
                primary: "1234"
           }
}`

as you have used $set, after updating the secondary number will be gone

Reply:
I already assign the entire customer object in the server layer & and just changed the password. So here this function is taking the full customer info.